src/GafferTest/ContextTest.cpp

```diff
@@ -206,6 +207,67 @@ void GafferTest::testEditableScope()
 
 }
 
+// Create the number of contexts specified, and return a count for how many collisions there are
+// in the last 32 bits of the context hash.  The uniformity of the last 32 bits should be
+// representative of the uniformity of the hash as a whole ( we don't want to check for collisions
+// in the whole 128 bit hash, since it would take years for one to occur randomly )
+// "mode" switches betwen 4 modes for creating contexts:
+//   0 :  1 entry with a single increment int
+//   1 :  40 fixed strings, plus a single incrementing int
+//   2 :  20 random floats
+//   3 :  an even mixture of the previous 3 modes
+// "chunk" is set to 0-3 to set which of the four 32 bit chunks of the hash to take, from the lowest 32
+//   bits to the highest 32 bits
+// "seed" can be used to perform different runs to get an average number.
+// The goal is that regardless of how we create the contexts, they are all unique, and should therefore
+// have an identical chance of collisions if our hashing performs ideally.
+int GafferTest::countContextHash32Collisions( int contexts, int mode, int chunk, int seed )
+{
+	std::unordered_set<uint32_t> used;
+
+	srand( seed );
```

Comment:
This isn't thread-safe, and we do seem to be calling this function multiple times in parallel. Perhaps `rand_r()` or something out of `https://en.cppreference.com/w/cpp/numeric/random`.

Reply:
Oh, good point.

Reply:
Turned out to be a very good point: in combination with the precreation of InternedStrings mentioned below, switching to rand_r() means this test is now only taking 70 seconds, and the distribution does not seem to have been affected.

Reply:
Fixed in 717b9a8